io/net/Network_config.py

```diff
@@ -0,0 +1,127 @@
+                for val in tmp:
+                    if 'Kernel driver in use' in val:
+                        cmd, driverinfo = val.split(': ')
+                        self.log.info(driverinfo)
+                        if driver != driverinfo:
+                            self.fail("mismatch in driver information")
+        cmd = "cat /sys/module/%s/drivers/pci\:%s/%s/net/%s/mtu" %\
+              (driver, driver, businfo, self.iface)
+        mtu = process.system_output(cmd, shell=True).strip()
+        self.log.info("mtu value is %s" % mtu)
+        cmd = "cat /sys/module/%s/drivers/pci\:%s/%s/net/%s/operstate" %\
+              (driver, driver, businfo, self.iface)
+        operstate = process.system_output(cmd, shell=True).strip()
+        self.log.info("operstate is %s" % operstate)
+        cmd = "cat /sys/module/%s/drivers/pci\:%s/%s/net/%s/duplex" %\
+              (driver, driver, businfo, self.iface)
+        duplex = process.system_output(cmd, shell=True).strip()
+        self.log.info("transmission mode is %s" % duplex)
+        cmd = "cat /sys/module/%s/drivers/pci\:%s/%s/net/%s/address" %\
+              (driver, driver, businfo, self.iface)
+        address = process.system_output(cmd, shell=True).strip()
+        self.log.info("mac address is %s" % address)
+        cmd = "cat /sys/module/%s/drivers/pci\:%s/%s/net/%s/speed" %\
+              (driver, driver, businfo, self.iface)
+        speed = process.system_output(cmd, shell=True).strip()
+        self.log.info("speed is %s" % speed)
+        cmd = "ethtool %s | grep Speed | awk '{print $2}'" % self.iface
+        eth_speed = process.system_output(cmd, shell=True).strip()
+        eth_speed = eth_speed.strip('Mb/s')
+        self.log.info("Performing Ethtool and interface checks for interface")
+        self.log.info("-----------------------------------------------------")
+        if speed != eth_speed:
+            self.fail("mis match in speed")
+        hw_addr = netifaces.ifaddresses(self.iface)[netifaces.AF_LINK]
+        hw_addr = hw_addr[0]['addr']
+        if hw_addr != address:
+            self.fail("mismatch in hardware address")
+        cmd = "ifconfig %s" % self.iface
+        mtuval = process.system_output(cmd, shell=True).strip()
+        tmp = mtuval.split('\n')
+        for val in tmp:
+            if 'mtu' in val:
+                cmd, mtu_val = val.split('mtu ')
+                self.log.info("through ifconfig mtu value is %s" % mtu_val)
+                if mtu != mtu_val:
+                    self.fail("mismatch in mtu")
+            if 'MTU' in val:
+                cmd, mtu_val = val.split('MTU:')
+                mtu_val, cmd = mtu_val.split(' ')
+                self.log.info("through ifconfig mtu value is %s" % mtu_val)
+                if mtu != mtu_val:
+                    self.fail("mismatch in mtu")
+        eth = "ethtool %s | grep 'Link detected:'" % self.iface
+        eth_state = process.system_output(eth, shell=True)
```

Comment:
here as well can this be 

`eth_state = process.system_output("ethtool %s | grep 'Link detected:'" % self.iface, shell=True)`